src/main/java/org/skriptlang/skript/lang/comparator/Comparators.java

```diff
@@ -44,17 +44,16 @@ private Comparators() {}
 	/**
 	 * A default comparator to compare two objects using {@link Object#equals(Object)}.
 	 */
-	public static final Comparator<Object, Object> EQUALS_COMPARATOR = (o1, o2) -> Relation.get(o1.equals(o2));
-
-	/**
-	 * A Java {@link java.util.Comparator} for comparing two objects using {@link #compare(Object, Object)}.
-	 */
-	public static final java.util.Comparator<Object> JAVA_COMPARATOR = (o1, o2) -> compare(o1, o2).getRelation();
+	private static final ComparatorInfo<Object, Object> EQUALS_COMPARATOR_INFO = new ComparatorInfo<>(
+		Object.class,
+		Object.class,
+		(o1, o2) -> Relation.get(o1.equals(o2))
+	);
 
 	/**
 	 * A List containing information for all registered comparators.
 	 */
-	private static final List<ComparatorInfo<?, ?>> COMPARATORS = Collections.synchronizedList(new ArrayList<>());
+	private static final List<ComparatorInfo<?, ?>> COMPARATORS = Collections.synchronizedList(new ArrayList<>(50));
 
 	/**
 	 * @return An unmodifiable, synchronized list containing all registered {@link ComparatorInfo}s.
```

Comment:
Wouldn't it be better to return a copy of the list instead?

Reply:
Not sure - do we want to return a modifiable list?

Reply:
Don't think we should leave sth like this up to the caller (i.e. allow modification), no idea what would be done in that case (think of removal, adding null values, duplicate signatures etc). 

My concern with unmodifiableList is that it doesn't return a copy, it returns a 'live' unmodifiable view. So if the original is updated, so will the view. If the an iterator is created from the view, and the original is modified, continuing to use the iterator will throw concurrent modification exception

Reply:
It should either return a copy or everything gets replaced with declarative style (e.g. `Comparators#forEach(Consumer)`, `Comparators#map(Function)`)

Reply:
The unmodifiable view should be good here (over the copy) as the list is guaranteed to never change after registration (and the method is not available until after registration). I figured a copy would likely be wasteful because of this. Feel free to comment/unresolve if you still have concerns.